xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -772,7 +772,11 @@ void CGUIWindowManager::ActivateWindow(int iWindowID, const std::vector<std::str
   {
     // make sure graphics lock is not held
     CSingleExit leaveIt(CServiceBroker::GetWinSystem()->GetGfxContext());
-    CApplicationMessenger::GetInstance().SendMsg(TMSG_GUI_ACTIVATE_WINDOW, iWindowID, swappingWindows ? 1 : 0, nullptr, "", params);
+    CGUIWindow* pWindow = GetWindow(iWindowID);
+    if (!pWindow || pWindow->IsDialog())
```

Comment:
I don't know if this is actually needed, maybe someone can tell me if we should block till the window (if not a dialog) is loaded.